### PR TITLE
 Apply Node.js service restructure to pong

### PR DESCRIPTION
**Pull request by bbridges**
### Changes

- Added changes brought to forward-interop and telemetry to the pong
  service.
- Using a class to store the ping data.
- Using only environment variables to configure service instead of using
  the command as well.
- No longer marking the pong service as an allowed failure on Travis CI.
- Updated the test configuration.
- Removing protoc as a dependency of pong.

### Tests

The same test case as in the previous version is essentially the same, but ported to jest and using nock to mock the test APIs instead.

Also tested manually by using the testing configuration in `test/` by looking at the pong service display on dashboard and tested it by killing off a service and seeing it went offline for a bit until it came back.

### CODE REVIEW

File: services/pong/README.md [hunk 1]
```diff
@@ -3,22 +3,16 @@
...
+## Environment Variables
+
+- `SERVICE_TIMEOUT` - amount of time in ms until a request times out - defaults
+  to 5000 ms.
+- `PING_SERVICES` - list of services to ping in the form
+  `name,host:port,endpoint` and separated by spaces. Port and endpoint are
+  optional.
+
+  For example:
+  `some-service,192.168.0.3 another-one,interop:1234,/endpoint`
```
Comment (oldmud0) [inline services/pong/README.md]: Still a little unclear how this works.

File: services/pong/src/ping-store.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    });
+  }
+
+  /**
+   * Feed new service ping to the store.
+   *
+   * @param {string}  name
+   * @param {boolean} online
+   * @param {number}  ms
+   */
```
Comment (oldmud0) [inline services/pong/src/ping-store.js]: `ms`? Oh, milliseconds...

File: services/pong/src/service.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        // being successful.
+        .redirects(0)
+        .ok(res => res.status < 400);
+
+      const time = Date.now() - start;
+
+      // Mark the service's ping time on success.
+      this._pingStore.updateServicePing(service, true, time);
+    } catch (err) {
+      // FIXME: Check for what kind of error this is.
```
Comment (oldmud0) [inline services/pong/src/service.js]: Remember you still have a `FIXME` here

Comment (bbridges) [reply services/pong/src/service.js]: Yep, will add a story afterwards to go back here and handle this in a better way.

File: services/pong/src/ping-store.js (hunk 2 above)
Comment (bbridges) [reply services/pong/src/ping-store.js]: `ms` is what it's called in the proto file so I kept it like that there.

File: services/pong/README.md (hunk 1 above)
Comment (bbridges) [reply services/pong/README.md]: Would describing it more help, or adding more examples?

File: services/pong/src/service.js (hunk 3 above)
Comment (bbridges) [reply services/pong/src/service.js]: Actually, I can make the `try` block only consist of the request and that should be fine.

File: services/pong/README.md (hunk 1 above)
Comment (oldmud0) [reply services/pong/README.md]: A better example would be nice - probably just take whatever is currently in the `docker-compose.yml`.

**Pull request merged**